File: algorithms/python/maximumNumberofBalloons/maximumNumberofBalloons.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/maximum-number-of-balloons/
# Author: Miao Zhang
# Date:   2021-04-16

class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        cnt = collections.Counter(text)
        return min(cnt['b'], cnt['a'], cnt['l'] // 2, cnt['o'] // 2, cnt['n'])

File: algorithms/cpp/maximumNumberofBalloons/maximumNumberofBalloons.cpp
// Source: https://leetcode.com/problems/maximum-number-of-balloons/
// Author: Miao Zhang
// Date:   2021-04-16

class Solution {
public:
    int maxNumberOfBalloons(string text) {
        map<char, int> cnt;
        for (char t: text) cnt[t]++;
        return min({cnt['b'], cnt['a'], cnt['l'] / 2, cnt['o'] / 2, cnt['n']});
    }
};
